tests/components/onewire/test_entity_sysbus.py

```diff
@@ -1,129 +1,121 @@
+        "sensors": [],
+    },
+    "EF-111111111112": {
+        "sensors": [],
+    },
+}
+
+
+@pytest.mark.parametrize("device_id", MOCK_DEVICE_SENSORS.keys())
+async def test_onewiredirect_setup_valid_device(hass, device_id):
     """Test that sysbus config entry works correctly."""
     entity_registry = mock_registry(hass)
-    config = {
-        "sensor": {
-            "platform": DOMAIN,
-            "mount_dir": DEFAULT_SYSBUS_MOUNT_DIR,
-            "port": DEFAULT_OWSERVER_PORT,
-            "names": {
-                MOCK_DEVICE_ID: MOCK_DEVICE_NAME,
-            },
-        }
-    }
+
+    read_side_effect = []
+    expected_sensors = MOCK_DEVICE_SENSORS[device_id]["sensors"]
+    for expected_sensor in expected_sensors:
+        read_side_effect.append(expected_sensor["injected_value"])
 
     with patch(
         "homeassistant.components.onewire.sensor.Pi1Wire"
     ) as mock_pi1wire, patch("pi1wire.OneWire") as mock_owsensor:
         type(mock_owsensor).mac_address = PropertyMock(
-            return_value=MOCK_DEVICE_ID.replace("-", "")
+            return_value=device_id.replace("-", "")
         )
-        mock_owsensor.get_temperature.side_effect = [
-            25.123,
-            FileNotFoundError,
-            25.223,
-            InvalidCRCException,
-            25.323,
-            NotFoundSensorException,
-            25.423,
-            UnsupportResponseException,
-            25.523,
-        ]
+        mock_owsensor.get_temperature.side_effect = read_side_effect
         mock_pi1wire.return_value.find_all_sensors.return_value = [mock_owsensor]
-        assert await async_setup_component(hass, SENSOR_DOMAIN, config)
-        await hass.async_block_till_done()
-
-        assert len(entity_registry.entities) == 1
-        registry_entry = entity_registry.entities.get(MOCK_ENTITY_ID)
-        assert registry_entry is not None
-        assert (
-            registry_entry.unique_id == f"/sys/bus/w1/devices/{MOCK_DEVICE_ID}/w1_slave"
-        )
-        assert registry_entry.unit_of_measurement == TEMP_CELSIUS
-
-        # 25.123
-        current_time = datetime.now()
-        state = hass.states.get(MOCK_ENTITY_ID)
-        assert state.state == "25.1"
-
-        # FileNotFoundError
-        current_time = current_time + timedelta(minutes=2)
-        async_fire_time_changed(hass, current_time)
-        await hass.async_block_till_done()
-        state = hass.states.get(MOCK_ENTITY_ID)
-        assert state.state == "unknown"
-
-        # 25.223
-        current_time = current_time + timedelta(minutes=2)
-        async_fire_time_changed(hass, current_time)
-        await hass.async_block_till_done()
-        state = hass.states.get(MOCK_ENTITY_ID)
-        assert state.state == "25.2"
-
-        # InvalidCRCException
-        current_time = current_time + timedelta(minutes=2)
-        async_fire_time_changed(hass, current_time)
+        assert await async_setup_component(hass, SENSOR_DOMAIN, MOCK_CONFIG)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)